Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -662,19 +664,19 @@ public static ActionAssertions Should(this Action action)
         /// current <see cref="System.Func{Task}"/> .
         /// </summary>
         [Pure]
-        public static AsyncFunctionAssertions Should(this Func<Task> action)
+        public static NonGenericAsyncFunctionAssertions Should(this Func<Task> action)
```

Comment:
Strictly speaking this is apparently a breaking change
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#signatures

but those rules are also ultra strict as they need to be reflection and serialization compatible.
Since we're returning a more derived type, we should be covered?

Reply:
From a compile-time perspective, it'll be fine indeed. But technically, this could cause a binary incompatibility of some other library relied on this method. I'm not sure I'm willing to be that strict.

Reply:
Do you agree with this?

Reply:
Wouldn't that other library be able to match it against the base type `AsyncFunctionAssertions`?
Maybe I'm just to tired to review any more code for today.

Reply:
I was to apparently to tired to even answer your question.
I'm good with this change and the potential risk.